### ISSUE TITLE
setValueCurveAtTime when curve has non-finite values?

### ISSUE THREAD

**Issue by rtoy** [WebAudio/web-audio-api#1171]
The spec currently doesn't say anything about  the values contained in the curve for `setValueCurveAtTime`.  Are we allowing any possible value for the curve, including NaN and Infinity?

Chrome currently signals an error if the curve contains non-finite values.

Allowing such values would probably very quickly cause NaN to be produced for the automation, probably also causing all downstream nodes to produce NaN for all audio samples.  Figuring out why this happened is fairly hard.

**Comment by padenot** [comment]
I think this is taken care of already. Allowing `NaN` and `Infinity` in the ctor would be written as:

```
sequence<unrestricted float> imag;
```.

We might want to change the factory method so that it uses `sequence<float>` as well ?

**Comment by rtoy** [comment]
Yes, I think using `sequence<float>` everywhere instead of `Float32Array` would be a nice convenience.

**Comment by padenot** [comment]
Yeah. What would it mean in terms of breakage ? I think it would
work fine.

**Comment by rtoy** [comment]
Can't think of any cases where `sequence<float>` would break anything.  Not sure if Chrome's IDL compiler will actually check to see of the sequence only contained finite values, though.  But that's an implementation issue, not a spec issue.

**Comment by joeberkovitz** [comment]
`sequence<float>` is OK. We need to prevent propagating NaN/Infinity values.

**Comment by rtoy** [comment]
Heh. This is a duplicate of #815, fixed by #819. But we seem to have lost the text about throwing errors.  And #815 mentioned using `sequence<float>` but we never did.

Let's actually make it happen this time, with `sequence<float>`.
